### PR TITLE
Fix parametric automaton transitions

### PR DESCRIPTION
**Pull request by Pavinati**
When evaluating the transitions of a placeholder token, take all the possible transitions for that state

### CODE REVIEW

File: lib/astarte_core/mapping.ex [hunk 1]
```diff
@@ -148,6 +148,14 @@ defmodule Astarte.Core.Mapping do
     String.replace(endpoint, @placeholder_regex, "")
   end
 
+  @doc """
+  Check if token is a placeholder.
+  """
+  @spec is_placeholder?(String.t()) :: boolean()
+  def is_placeholder?(token) when is_binary(token) do
+    Regex.run(@placeholder_regex, token) != nil
```
Comment (rbino) [inline lib/astarte_core/mapping.ex]: Use `String.match(...)`

File: lib/astarte_core/mapping/endpoints_automaton.ex [hunk 2]
```diff
@@ -148,6 +147,17 @@ defmodule Astarte.Core.Mapping.EndpointsAutomaton do
     end
   end
 
+  defp state_transitions(transitions, state) do
+    Enum.reduce(transitions, [], fn transition, acc ->
+      if match?({{^state, _}, _}, transition) do
```
Comment (rbino) [inline lib/astarte_core/mapping/endpoints_automaton.ex]:
```
fn {{^state, _}, _}, transition ->
  ...
fn _, _ ->
  acc

File: lib/astarte_core/mapping/endpoints_automaton.ex [hunk 3]
```diff
@@ -93,24 +93,30 @@ defmodule Astarte.Core.Mapping.EndpointsAutomaton do
...
+            end
+
+          epsi_transition_list =
+            if epsi_transition do
+              [epsi_transition]
+            else
+              []
+            end
+
+          transition_list ++ epsi_transition_list ++ acc
```
Comment (rbino) [inline lib/astarte_core/mapping/endpoints_automaton.ex]:
```elixir
cond do
  is_nil(transition) and is_nil(epsi_transition) ->
    acc
  is_nil(transition_list) ->
    [epsi_transition | acc]
  is_nil(epsi_transition) ->
    [transition_list | acc]
  true ->
    [transition, epsi_transition | acc]
end

File: lib/astarte_core/mapping/endpoints_automaton.ex [hunk 4]
```diff
@@ -93,24 +93,30 @@ defmodule Astarte.Core.Mapping.EndpointsAutomaton do
...
-            [epsi_transition]
-          else
-            []
-          end
-
-        transition_list ++ epsi_transition_list ++ acc
+          all_state_transitions ++ acc
+        else
+          transition = Map.get(transitions, {state, token})
+          epsi_transition = Map.get(transitions, {state, ""})
```
Comment (rbino) [inline lib/astarte_core/mapping/endpoints_automaton.ex]:
```elixir
transitions_list = Map.get(...) |> List.wrap()
```

File: lib/astarte_core/mapping/endpoints_automaton.ex [hunk 5]
```diff
@@ -93,24 +93,30 @@ defmodule Astarte.Core.Mapping.EndpointsAutomaton do
...
+              []
+            end
+
+          epsi_transition_list =
+            if epsi_transition do
+              [epsi_transition]
+            else
+              []
+            end
+
```
Comment (rbino) [inline lib/astarte_core/mapping/endpoints_automaton.ex]: And remove this

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,9 +5,15 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [0.11.0-beta.3] - Unreleased
+### Added
+- Additional overlapping endpoint tests.
```
Comment (bettio) [inline CHANGELOG.md]: Remove this

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,9 +5,15 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+
 ### Changed
 - Add support for aggregated server owned interfaces.
 
+### Fixed
+- Known issue with overlapping endpoints #2.
```
Comment (bettio) [inline CHANGELOG.md]: "Invalid overlapping endpoints, now they are always refused."

Comment (rbino) [reply CHANGELOG.md]: Correctly handle parametric endpoints regardless of the ordering, so that overlapping endpoints are always refused. (See #2)

**Pull request merged**